File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

class Gigasecond {

    private final LocalDateTime birthDateTime;

    Gigasecond(LocalDate birthDate) {
        birthDateTime = birthDate.atStartOfDay();
    }

    Gigasecond(LocalDateTime birthDateTime) {
        this.birthDateTime = birthDateTime;
    }

    LocalDateTime getDate() {
        return birthDateTime.plusSeconds(1_000_000_000);
    }

}

File: kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.temporal.ChronoUnit.*

class Gigasecond(startTime: LocalDateTime) {
    private val amount = Math.pow(10.0, 9.0).toLong()

    val date: LocalDateTime = startTime.plus(amount, SECONDS)

    constructor(startDate: LocalDate) : this(startDate.atStartOfDay())

}